service/src/main/kotlin/fi/espoo/evaka/messaging/bulletin/BulletinQueries.kt

```diff
@@ -0,0 +1,173 @@
+package fi.espoo.evaka.messaging.bulletin
+
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.NotFound
+import fi.espoo.evaka.shared.utils.zoneId
+import org.jdbi.v3.core.kotlin.mapTo
+import java.time.OffsetDateTime
+import java.util.UUID
+
+fun Database.Transaction.initBulletin(
+    user: AuthenticatedUser
+): UUID {
+    // language=sql
+    val sql = """
+        INSERT INTO bulletin (created_by_employee)
+        VALUES (:createdBy)
+        RETURNING id
+    """.trimIndent()
+    return this.createQuery(sql)
+        .bind("createdBy", user.id)
+        .mapTo<UUID>()
+        .first()
+}
+
+fun Database.Transaction.updateDraftBulletin(
+    user: AuthenticatedUser,
+    id: UUID,
+    groupId: UUID?,
+    title: String,
+    content: String
+) {
+    // language=sql
+    val sql = """
+        UPDATE bulletin
+        SET group_id = :groupId, title = :title, content = :content
+        WHERE id = :id AND created_by_employee = :userId
+    """.trimIndent()
+
+    val updated = this.createUpdate(sql)
+        .bind("id", id)
+        .bind("userId", user.id)
+        .bind("groupId", groupId)
+        .bind("title", title)
+        .bind("content", content)
+        .execute()
+
+    if (updated == 0) throw NotFound("No bulletin $id found by user ${user.id} in draft state")
```

Comment:
You don't actually check that the bulletin is a draft in the update, so this isn't completely accurate.

Reply:
oh yeah, fixed